orb_models/dataset/ase_dataset.py

```diff
@@ -0,0 +1,258 @@
+from pathlib import Path
+from typing import Dict, Optional, Tuple, Union
+
+import ase
+import ase.db
+import ase.db.row
+import numpy as np
+import torch
+from ase.stress import voigt_6_to_full_3x3_stress
+from e3nn import o3
+from torch.utils.data import Dataset
+
+from orb_models.forcefield import atomic_system, property_definitions
+from orb_models.forcefield.base import AtomGraphs
+
+
+class AseSqliteDataset(Dataset):
+    """AseSqliteDataset.
+
+    A Pytorch Dataset for reading ASE Sqlite serialized Atoms objects.
+
+    Args:
+        name: The dataset name.
+        path: Local or GCS path to the sqlite file to read, or to a directory containing .db files
+            representing shards of a dataset. Shards are read in alphabetically sorted
+            order, so it is important to name them in a way that preserves the order
+            of the dataset.
+        system_config: A config for controlling how an atomic system is represented.
+        target_config: A config for regression/classification targets.
+        augmentation: If random rotation augmentation is used.
+
+    Returns:
+        An AseSqliteDataset.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        path: Union[str, Path],
+        system_config: Optional[atomic_system.SystemConfig] = None,
+        target_config: Optional[atomic_system.PropertyConfig] = None,
+        augmentation: Optional[bool] = True,
+    ):
+        super().__init__()
+        self.name = name
+        self.augmentation = augmentation
+        self.path = path
+        self.db = ase.db.connect(str(self.path), serial=True, type="db")
+
+        self.feature_config = system_config
+        self.target_config = target_config
+
+    def __getitem__(self, idx) -> AtomGraphs:
+        """Fetch an item from the db.
+
+        Args:
+            idx: An index to fetch from the db file and convert to an AtomGraphs.
+
+        Returns:
+            A AtomGraphs object containing everything the model needs as input,
+            positions and atom types and other auxillary information, such as
+            fine tuning targets, or global graph features.
+        """
+        # Sqlite db is 1 indexed.
+        row = self.db.get(idx + 1)
+        atoms = row.toatoms()
+        extra_feats = self._get_row_properties(row, self.feature_config)
+        extra_targets = self._get_row_properties(row, self.target_config)
```

Comment:
We only need to support the following targets:
- forces
- stress
- energy

Can we simplify this code such that we don't need the target config, and instead we pass a list of these keys to the dataset class, specifying which ones we expect to be present.